pkg/nfd-master/nfd-master.go

```diff
@@ -346,6 +388,85 @@ func removeLabels(n *api.Node, labelNames []string) {
 	}
 }
 
+func patchNodeStatus(marshalable interface{}, nodeName string, helper apihelper.APIHelpers) error {
+	patch, err := json.Marshal(marshalable)
+	if err == nil {
+		cli, err := helper.GetClient()
+		if err != nil {
+			return err
+		}
+		err = helper.PatchStatus(cli, nodeName, patch)
+	}
+	return err
+}
+
+// pruneExtResources forms a patch which removes redundant ext resources from
+// node status capacity and allocatable arrays
+// returns true if node was patched, false otherwise (and error)
+func pruneExtResources(n *api.Node, labels Labels, resourceLabels ResourceLabels, oldResources []string, helper apihelper.APIHelpers) (bool, error) {
```

Comment:
I think you could combine `pruneExtResources()` and `addExtResources()` into one `updateExtendedResources()` function. 
```
func updateExendedtResources(n *api.Node, extendedResources ExtendedResources, oldResources []string, helper apihelper.APIHelpers)
```

Create a list of patches, doing "replace" for existing resources that needs to be updated, "add" for completely new resources and "remove" for resources to be removed. Then do a atomic update on one `patchNodeStatus()` call